src/scenes/Login/Login.js

```diff
@@ -1,12 +1,15 @@
 import React, { useState } from 'react'
 import './_login.scss'
 import PropTypes from 'prop-types'
+import { getToken, saveToken } from '../../useToken'
 
-const Login = ({ history }) => {
+const Login = () => {
+  if (getToken()) {
+    return window.location.replace('/')
+  }
```

Comment:
if you put the check here, `getToken()` will be called a lot of times. Especially when you typing and the form is re-rendered. You may want to verify it.
So you need to find a solution to avoid this performance issue. Think about when you need to check the token and how many times it should be called.